src/deluge/gui/ui/save/save_song_ui.cpp

```diff
@@ -237,7 +237,7 @@ bool SaveSongUI::performSave(StorageManager& bdsm, bool mayOverwrite) {
 				// converted
 
 				// Open file to read
-				FRESULT result = f_open(&fileSystemStuff.currentFile, sourceFilePath, FA_READ);
+				FRESULT result = f_open(&smSerializer.writeFIL, sourceFilePath, FA_READ);
```

Comment:
Same here - I think we should keep the actual file system calls centralized

Reply:
Also handled by above change.